### PR TITLE
Add create_ek() to tpm.rs

### PR DESCRIPTION
**Pull request by lkatalin**
If this needs more functionality, different return types, etc., please let me know.

### CODE REVIEW

File: src/tpm.rs [hunk 1]
```diff
@@ -24,3 +25,36 @@ pub(crate) fn get_tpm2_ctx() -> Result<tss_esapi::Context> {
...
+ * let (key, cert) = tpm::create_ek(context, Some(AsymmetricAlgorithm::Rsa))
+ */
+pub(crate) fn create_ek(
+    context: &mut Context,
+    alg: Option<AsymmetricAlgorithm>,
+) -> Result<(KeyHandle, Vec<u8>)> {
+    let alg = match alg {
+        Some(a) => a,
+        None => {
+            // TODO: What other values could be in keylime.conf?
```
Comment (puiterwijk) [inline src/tpm.rs:42]: I think the only other value we should consider correct is `ecc`.
I'd say that the `_ =>` should return an error instead.

**Review by puiterwijk — COMMENTED** [review]
This looks decent with just one small comment!

**Comment by lkatalin** [comment]
Here is an updated PR with an error for encryption algorithms other than rsa and ecc, and retrieving the TPM pub object and turning it into a `Vec<u8>`.

I don't think that turning the TPM pub object into a `Vec<u8>` should return `KeylimeError::InvalidRequest`; it seems better to return a `TPM(tss_esapi::Error::WrapperError(...WrapperErrorKind::WrongValueFromTPM))`, but I wasn't able to create one because it would not let me import `WrapperErrorKind`. Any thoughts on the best Error type here? If we end up just using an error from Keylime, I will drop the import of the `tss_esapi::Error`.

**Comment by lkatalin** [comment]
Depends on #145 #144

**Review by puiterwijk — APPROVED** [review]

**Pull request merged**